src/tests.rs

```diff
@@ -241,131 +239,90 @@ fn get_chunk_sizes() -> Result<(), Error> {
-    let subset: Vec<_> = encrypted_chunks
-        .iter()
-        .filter(|c| c.index >= info.index_range.start && c.index <= info.index_range.end)
-        .sorted_by_key(|c| c.index)
-        .cloned()
-        .collect();
-
-    let read_data = decrypt_range(data_map, &subset, info.relative_pos, len)?;
-
-    compare(expected_data, read_data.clone())?;
-
-    Ok(read_data)
+#[test]
+fn seek_with_length_over_data_size() -> Result<(), Error> {
+    let file_size = 10_000_000;
+    let bytes = random_bytes(file_size);
+    let start_pos = 512;
+
+    // Calculate length safely
+    let remaining_bytes = file_size.saturating_sub(start_pos);
+    let len = remaining_bytes.saturating_add(1); // Try to read one more byte than available
+
+    let (data_map, encrypted_chunks) = encrypt_chunks(bytes.clone())?;
+
+    // We expect to get data from start_pos to end of file
+    let expected_data = bytes.slice(start_pos..file_size);
+
+    let read_data = decrypt_range(&data_map, &encrypted_chunks, start_pos, len)?;
+    compare(expected_data, read_data)?;
+
+    // Also verify reading beyond end returns empty
+    let read_data = decrypt_range(&data_map, &encrypted_chunks, file_size + 1, 1)?;
+    assert!(
+        read_data.is_empty(),
+        "Reading beyond end should return empty"
+    );
+
+    Ok(())
 }
 
 #[test]
 fn seek_over_chunk_limit() -> Result<(), Error> {
-    // Having first chunk being at index 1 starts at position: 4_194_304
     let start_size = 4_194_300;
-    for i in 0..27 {
+    for i in 0..5 {
+        // Reduced iterations
         let file_size = start_size + i;
         let bytes = random_bytes(file_size);
-
         let pos = file_size / 4;
-        let len = file_size / 2;
+        let len = std::cmp::min(file_size / 2, file_size - pos); // Ensure we don't read past end
 
-        // this is what we expect to get back from the chunks
         let expected_data = bytes.slice(pos..(pos + len));
-
-        // the chunks covering the bytes we want to read
-        let (start_index, end_index) = overlapped_chunks(file_size, pos, len);
-
-        // first encrypt the whole file
         let (data_map, encrypted_chunks) = encrypt_chunks(bytes.clone())?;
 
-        // select a subset of chunks; the ones covering the bytes we want to read
-        let subset: Vec<_> = encrypted_chunks
-            .into_iter()
-            .filter(|c| c.index >= start_index && c.index <= end_index)
-            .sorted_by_key(|c| c.index)
-            .collect();
-
-        // the start position within the first chunk (thus `relative`..)
-        let relative_pos = pos % get_chunk_size(file_size, start_index);
-        let read_data = decrypt_range(&data_map, &subset, relative_pos, len)?;
-
+        let read_data = decrypt_range(&data_map, &encrypted_chunks, pos, len)?;
         compare(expected_data, read_data)?;
     }
-
-    Ok(())
-}
-
-#[test]
-fn seek_with_length_over_data_size() -> Result<(), Error> {
```

Comment:
better keep this scenario (may can be merged with above)

Reply:
This test used old mechanism and was updated to the new code in a new test. We could do another larger test again, but this suite was upgraded to handle new code.